src/init.cpp

```diff
@@ -1301,7 +1301,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         }
     }
 
-    for (const auto &port_option : std::vector<std::pair<std::string, bool>>{
+    for (const auto& [arg, unix] : std::vector<std::pair<std::string, bool>>{
```

Comment:
ACK on improving the syntax, but it seems a matter of time before the compiler again becomes smart enough to see this variable is unused?
Mayb eadd
```
(void)unix;
```
to the `!HAVE_SOCKADDR_UN` path just in case?